wordpress-mobile/WordPress-FluxC-Android
# [Woo] Coupons: Save `woocommerce_enable_coupons` value from the response of `/settings/general` API call.

### PR Description

The response from `wcCoreRestClient.fetchSiteSettingsGeneral()` has a `woocommerce_enabled_coupons` field that currently isn't saved/used in FluxC ([check here for example](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/be807d3717b0b0b9b4c812a65026504096d5615e/example/src/androidTest/resources/wc-site-settings-general-response-success.json#L233-L253)). 

This PR adds saving the value, so that the WooCommerce app can make use of it to check whether Coupons is enabled or not on the site.

### To Test:
1. Run Example app,
2. Go to Woo, and "Select Site",
3. Click "Fetch WC Site Settings",
4. Observe the result and ensures it includes `couponsEnabled=1` or `couponsEnabled=0`.
